shell/src/main/java/org/apache/accumulo/shell/Shell.java

```diff
@@ -261,12 +282,27 @@ public Shell(LineReader reader) {
     this.writer = terminal.writer();
   }
 
+  // this is visible only for FateCommandTest, otherwise, should be private or inline
+  protected boolean initialAuthentication(AccumuloClient accumuloClient, AuthenticationToken token)
+      throws AccumuloException, AccumuloSecurityException {
+    final Credentials toAuth = new Credentials(accumuloClient.whoami(), token);
+    return ThriftClientTypes.SHELL.execute(context, c -> c.authenticateUser(TraceUtil.traceInfo(),
```

Comment:
I talked with @EdColeman and he suggested that the Shell should be calling a method on SecurityOperations instead of this (which is functionally equivalent). This would either mean adding another method on SecurityOperations or modifying the existing SecurityOperations.authenticateUser method to do this. There is another option too which would potentially help all clients - move the background thread that populates ZooCache into AccumuloClient and modify the default implementation of `getTabletServerConnection` to use the logic in this PR. I'm going to put up a second PR to test this out.